### PR TITLE
add ordered command interface

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add the sequencing mode for commands

### Proposed changes

- add a sequencing mode parameter for commands
- add a "log XXXX" command (the part after the log gets put in the log of the target

### CODE REVIEW

File: src/helics/application_api/BrokerApp.hpp [hunk 1]
```diff
@@ -128,8 +128,12 @@ class HELICS_CXX_EXPORT BrokerApp {
    "federation",  "broker", "core", or the name of a specific object/core/broker
    @param target the specific target of the command
    @param commandStr the actual command
+   @param mode the ordering mode to use (fast for asynchronous priority channels, and ordered for
+    slower but well ordered queries)
```
Comment (nightlark) [inline src/helics/application_api/BrokerApp.hpp:132]:
```suggestion
    slower but well ordered commands)
```

File: src/helics/application_api/CoreApp.hpp [hunk 2]
```diff
@@ -130,8 +130,12 @@ class HELICS_CXX_EXPORT CoreApp {
    "federation",  "broker", "core", or the name of a specific object/core/broker
    @param target the specific target of the command
    @param commandStr the actual command
+    @param mode defaults to fast (asynchronous) meaning the query goes into priority channels,
+    ordered (synchronous) means slower queries but has more ordering guarantees
```
Comment (nightlark) [inline src/helics/application_api/CoreApp.hpp:134]:
```suggestion
    @param mode defaults to fast (asynchronous) meaning the command goes into priority channels,
    ordered (synchronous) means slower commands but has more ordering guarantees
```

File: src/helics/core/ActionMessageDefintions.hpp [hunk 3]
```diff
@@ -331,7 +333,9 @@ across different compilers*/
 #define CMD_SET_GLOBAL action_message_def::action_t::cmd_set_global
 
 #define CMD_SEND_COMMAND action_message_def::action_t::cmd_send_command
+#define CMD_SEND_COMMAND_ORDERED action_message_def::action_t::cmd_send_command_ordered
 #define CMD_COMMAND_STATUS action_message_def::action_t::cmd_command_status
+#define CMD_COMMAND_STATUS_ORDERED action_message_def::action_t::cmd_command_status
```
Comment (nightlark) [inline src/helics/core/ActionMessageDefintions.hpp:338]: Where is `cmd_command_status` defined? I'm not seeing it in the action_t enum in this file.

File: src/helics/core/Broker.hpp [hunk 4]
```diff
@@ -102,8 +102,12 @@ class Broker {
    "federation","root",  "broker", "core", or the name of a specific object/core/broker
    @param target the specific target of the command
    @param commandStr the actual command
+    @param mode fast (asynchronous; default) means the query goes on priority channels, ordered
+    (synchronous) is slower but has more ordering guarantees
```
Comment (nightlark) [inline src/helics/core/Broker.hpp:106]:
```suggestion
    @param mode fast (asynchronous; default) means the command goes on priority channels, ordered
    (synchronous) is slower but has more ordering guarantees
```

File: src/helics/core/Core.hpp [hunk 5]
```diff
@@ -773,10 +773,12 @@ class Core {
   @param target the specific target of the command
   @param commandStr the actual command
   @param source the designated source of the command, for return values or indication
+  @param mode the sequencing mode for the query, fast or ordered
```
Comment (nightlark) [inline src/helics/core/Core.hpp:776]:
```suggestion
  @param mode the sequencing mode for the command, fast or ordered
```

**Comment by phlptp** [comment]
Just for reference I am going to add some docs for the commands and tags in a coming soon PR.

**Pull request merged**